crates/biome_js_analyze/src/analyzers/style/use_template.rs

```diff
@@ -153,36 +153,43 @@ fn template_expression_from_binary_expression(
     // While `template_elements` is empty, we keep track of the last left node.
     // Once we see the first string/template literal,
     // we insert `last_left_node` in `template_elements` and the seen string/template literal.
-    // ANy subsequent expression is directly inserted in `template_elements`.
+    // Any subsequent expression is directly inserted in `template_elements`.
     let mut template_elements = vec![];
-    let mut last_left_node = None;
+    let mut left_nodes_stack = vec![];
     let mut iter = node.syntax().preorder();
     while let Some(walk) = iter.next() {
         match walk {
             WalkEvent::Enter(node) => {
                 let node = AnyJsExpression::cast(node)?;
                 match node {
-                    AnyJsExpression::JsParenthesizedExpression(_) => continue,
-                    AnyJsExpression::JsBinaryExpression(node)
-                        if matches!(node.operator().ok()?, JsBinaryOperator::Plus) =>
+                    AnyJsExpression::JsParenthesizedExpression(_) => {
+                        continue;
+                    }
+                    AnyJsExpression::JsBinaryExpression(ref binary)
+                        if matches!(binary.operator().ok()?, JsBinaryOperator::Plus)
+                            && evaluates_to_string(binary)? =>
```

Comment:
Similarly to `template_expression_from_binary_expression`, `evaluates_to_string` traverses the subtree. This duplicates some traversals. Do you think we could avoid double traversal?